skbio/metadata/_interval.py

```diff
@@ -0,0 +1,699 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import operator
+from functools import wraps
+
+from ._intersection import IntervalTree
+from skbio.util._decorator import experimental
+
+
+class Interval:
+    """Stores the location and metadata of an interval feature.
+
+    This class stores an interval feature. An interval feature
+    is defined as a sub-region of a sequence that is a functional
```

Comment:
Interval metadata can be applied to other types of objects, such as `TabularMSA`. Can you make this description a little more general? I really like the text here, so maybe rephrase this paragraph as an example of what interval metadata can be applied to (i.e. biological sequences)?